libraries/shared/watcher/storage_watcher.go

```diff
@@ -143,7 +144,11 @@ func (watcher StorageWatcher) transformDiff(diff types.PersistedDiff) error {
 
 	headerID, headerErr := watcher.getHeaderID(diff)
 	if headerErr != nil {
-		return fmt.Errorf("error getting header for diff: %w", headerErr)
+		if errors.Is(headerErr, ErrHeaderMismatch) {
+			return watcher.StorageDiffRepository.MarkNonCanonical(diff.ID)
```

Comment:
wondering if we may want to fire this in a goroutine in case it's time intensive 🤔